python/pyspark/sql/types.py

```diff
@@ -820,6 +837,9 @@ def from_ddl_datatype(type_str):
                 raise e
 
 
+DataType.fromDDL.__doc__ = _parse_datatype_string.__doc__
```

Comment:
Why do we do this?

Reply:
To avoid duplicating identical docs.  The better solution would be to eliminate _parse_datatype_string since DataType.fromDDL would be official public API and replace the need for it.  But I didn't want to break backward compatibility to quickly.

Perhaps a better compromise is to move the docs to DataType.fromDDL and then have _parse_datatype_string simply call DataType.fromDDL.  This cleans things up nicely and preserves backward compatibility.  If you like that plan I can make it happen.